Movie/Movie/MainModule/View/MovieViewController.swift

```diff
@@ -116,88 +96,54 @@ final class MovieViewController: UIViewController {
+    @objc private func refreshPageAction() {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            self.movieTableView.refreshControl?.beginRefreshing()
+            self.movieTableView.reloadData()
+            self.movieTableView.refreshControl?.endRefreshing()
         }
-        task.resume()
     }
 
     @objc private func changeSegmentAction(_ sender: UISegmentedControl) {
-        switch sender.selectedSegmentIndex {
-        case 0:
-            getMovies(genre: Constants.genres[0])
-        default:
-            getMovies(genre: Constants.genres[1])
-        }
+        presenter.updateSegmentControl(page: sender.selectedSegmentIndex)
     }
 }
 
-// MARK: - UITableViewDelegate, UITableViewDataSource
+// MARK: - UITableViewDataSource
 
-extension MovieViewController: UITableViewDelegate, UITableViewDataSource {
+extension MovieViewController: UITableViewDataSource {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        moviewDataSource?.results.count ?? 0
+        presenter.movies.count
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let data = moviewDataSource?.results[indexPath.row]
-        if indexPath.row == 0 {
-            let cell = tableView.dequeueReusableCell(
-                withIdentifier: Constants.adCellIdentifier,
-                for: indexPath
-            ) as? AdViewCell
-            cell?.setupUI(name: Constants.adImageName)
-            cell?.selectionStyle = .none
-            return cell ?? UITableViewCell()
-        } else {
-            let cell = tableView.dequeueReusableCell(
-                withIdentifier: Constants.movieCellIdentifier,
-                for: indexPath
-            ) as? MovieViewCell
-            cell?.selectionStyle = .none
-            guard let dataRelease = data?.releaseDate,
-                  let dataPosterPath = data?.posterPath,
-                  let dataTitle = data?.title,
-                  let dataVoteAverange = data?.voteAverage else { fatalError() }
-            let date = dateFormater.date(from: dataRelease)
-            dateFormater.dateFormat = Constants.editDateFormat
-            getMovieGenre(data: data)
-            let movie = Movies(
-                movieImageName: dataPosterPath,
-                movieGenreName: genres,
-                movieNameText: dataTitle,
-                movieDateText: dateFormater.string(from: date ?? Date()),
-                ratingValue: dataVoteAverange
-            )
-            self.movie = movie
-            cell?.setupView(movie: movie)
-            return cell ?? UITableViewCell()
-        }
+        let cell = tableView.dequeueReusableCell(
+            withIdentifier: Constants.movieCellIdentifier,
+            for: indexPath
+        ) as? MovieViewCell
+        cell?.selectionStyle = .none
+        cell?.setupView(movie: presenter.movies[indexPath.row], networkService: networkManager)
```

Comment:
```suggestion
        cell?.setupView(movie: presenter.movies[indexPath.row], networkService:)
```
 из презентера вытащить сервис